llvm/lib/CodeGen/ScheduleDAGInstrs.cpp

```diff
@@ -847,40 +932,9 @@ void ScheduleDAGInstrs::buildSchedGraph(AAResults *AA,
         (CanHandleTerminators || (!MI.isTerminator() && !MI.isPosition())) &&
         "Cannot schedule terminators or labels!");
 
-    // Add register-based dependencies (data, anti, and output).
-    // For some instructions (calls, returns, inline-asm, etc.) there can
-    // be explicit uses and implicit defs, in which case the use will appear
-    // on the operand list before the def. Do two passes over the operand
-    // list to make sure that defs are processed before any uses.
-    bool HasVRegDef = false;
-    for (unsigned j = 0, n = MI.getNumOperands(); j != n; ++j) {
-      const MachineOperand &MO = MI.getOperand(j);
-      if (!MO.isReg() || !MO.isDef())
-        continue;
-      Register Reg = MO.getReg();
-      if (Reg.isPhysical()) {
-        addPhysRegDeps(SU, j);
-      } else if (Reg.isVirtual()) {
-        HasVRegDef = true;
-        addVRegDefDeps(SU, j);
-      }
-    }
-    // Now process all uses.
-    for (unsigned j = 0, n = MI.getNumOperands(); j != n; ++j) {
-      const MachineOperand &MO = MI.getOperand(j);
-      // Only look at use operands.
-      // We do not need to check for MO.readsReg() here because subsequent
-      // subregister defs will get output dependence edges and need no
-      // additional use dependencies.
-      if (!MO.isReg() || !MO.isUse())
-        continue;
-      Register Reg = MO.getReg();
-      if (Reg.isPhysical()) {
-        addPhysRegDeps(SU, j);
-      } else if (Reg.isVirtual() && MO.readsReg()) {
-        addVRegUseDeps(SU, j);
-      }
-    }
+    /// Compute register dependences, calling back into reportDependence
+    /// whenever a dependence is detected.
```

Comment:
CHECK: I'm not sure whether reportDependence survived from the original refactoring